client/src/featureform/serving.py

```diff
@@ -609,6 +686,43 @@ def from_dataframe(dataframe, include_label_timestamp):
     def pandas(self):
         return self._dataframe
 
+    def wait(self, timeout=99999):
+        """Pauses execution and waits for trainingset metadata status to be set to ready before fetching data
+
+        **Examples**:
+        ``` py
+            client = ff.ServingClient()
+            dataset = client.training_set("fraud_training", "quickstart").wait(timeout=10)
+            # Waits 10 seconds before continuing. If 10 seconds elapses without training set setting to ready
+            # raise error. Otherwise continue
+            training_dataset = dataset.repeat(10) # Repeats data 10 times
+            for feature_batch in training_dataset:
+                # Train model
+        ```
+        Args:
+            timeout (int): The maximum number of seconds to wait before throwing error
+
+        Returns:
+            self (Dataset): Returns the current Dataset
+        """
+        if self._name is None or self._version is None:
+            raise ValueError("Local Dataset type does not implement wait")
+        training_set = self._resource_client.get_training_set(self._name, self._version, display=False)
+        timeout_duration = timedelta(seconds=timeout)
+        status = training_set.get_status()
+        time_waited = timedelta(seconds = 0)
+        time_started = datetime.now()
+        while (status != "FAILED" and status != "READY") and (timeout is None or time_waited < timeout_duration):
```

Comment:
can we move the wait logic into a separate helper function?

Reply:
moved

Reply:
Still here